dashboard/modules/metrics/install_and_start_prometheus.py

```diff
@@ -0,0 +1,192 @@
+import logging
+import os
+import platform
+import subprocess
+import sys
+import tarfile
+from pathlib import Path
+
+import requests
+
+from ray.dashboard.consts import PROMETHEUS_CONFIG_INPUT_PATH
+
+# Configure basic logging
+logging.basicConfig(
+    level=logging.INFO, format="%(asctime)s - %(levelname)s - %(message)s"
+)
+
+FALLBACK_PROMETHEUS_VERSION = "2.48.1"
+DOWNLOAD_BLOCK_SIZE = 8192  # 8 KB
+TEST_MODE_ENV_VAR = "RAY_PROMETHEUS_DOWNLOAD_TEST_MODE"
+
+
+def get_system_info():
+    os_type = platform.system().lower()
+    architecture = platform.machine()
+    if architecture == "x86_64":
+        # In the Prometheus filename, it's called amd64
+        architecture = "amd64"
+    return os_type, architecture
+
+
+def download_file(url, filename):
+    logging.info(f"Downloading {url} to {Path(filename).absolute()}...")
+    try:
+        test_mode = os.environ.get(TEST_MODE_ENV_VAR, False)
+        request_method = requests.head if test_mode else requests.get
+        response = request_method(url, stream=True)
+        response.raise_for_status()
+
+        total_size_in_bytes = int(response.headers.get("content-length", 0))
+        total_size_in_mb = total_size_in_bytes / (1024 * 1024)
+
+        downloaded_size_in_mb = 0
+        block_size = DOWNLOAD_BLOCK_SIZE
+
+        with open(filename, "wb") as file:
+            for chunk in response.iter_content(chunk_size=block_size):
+                file.write(chunk)
+                downloaded_size_in_mb += len(chunk) / (1024 * 1024)
+                print(
+                    f"Downloaded: {downloaded_size_in_mb:.2f} MB / "
+                    f"{total_size_in_mb:.2f} MB",
+                    end="\r",
+                )
+
+        print("\nDownload completed.")
```

Comment:
is there a higher level library for downloading byte data into files? 
This feels more manual than necessary.

I would think somethign like `library.download_file(url, out_path)` would already exist in one of the libraries we already have in our environment.

Reply:
There is [one used by Ray Data](https://filesystem-spec.readthedocs.io/en/latest/api.html#fsspec.implementations.http.HTTPFileSystem), but it doesn't provide a percent loading indicator, and I'd like to avoid the Ray Data dependency.  I think the percent indicator is useful otherwise it looks like the script is hanging (50mb takes seconds or 1-2 minutes depending on the internet speed).  I couldn't find a less manual way of getting a basic progress indicator.